### PR TITLE
Sell positions after some time

### PR DESCRIPTION
**Pull request by yaka996**
### PRs that wont be accepted
  - typos
  - editing comments (unless critical)
  - edits in documentation (unless critical)

  - [X] Functionality 
  - [ ] Ease of use   
  - [ ] Bug fix       

### How have you tested this PR

 - Validation must be ran in both `prod` and `test` (see TEST_MODE)
 - Enter a description here. Empty descriptions will be closed.

This has been tested in both test and prod. 
I haven't observed any issue so far.

### In your own words, please share how this makes the codebase better.

Added a new preference SELL_AFTER_DURATION to sell coins after a fixed duration in minutes.
Value 0 means this functionality is disabled.
For example, SELL_AFTER_DURATION: 15.0 means, that any coin will be sold if it's still held after 15 minutes from the time is was bought.

### CODE REVIEW

**Comment by ShogunMan** [comment]
Seems like a ok PR. Is it well tested?

**Comment by ShogunMan** [comment]
Looking at this again, if the pair is on the rise and increasing slowly it will still sell after x amount of time, you will then loose out of profits. Should be only for pairs that has not moved out of original TP.

File: Binance Detect Moonings.py [hunk 1]
```diff
@@ -378,9 +380,16 @@ def sell_coins():
             continue
 
         # check that the price is below the stop loss or above take profit (if trailing stop loss not used) and sell if this is the case
-        if LastPrice < SL or LastPrice > TP and not USE_TRAILING_STOP_LOSS:
-            print(f"{txcolors.SELL_PROFIT if PriceChange >= 0. else txcolors.SELL_LOSS}TP or SL reached, selling {coins_bought[coin]['volume']} {coin} - {BuyPrice} - {LastPrice} : {PriceChange-(TRADING_FEE*2):.2f}% Est:${(QUANTITY*(PriceChange-(TRADING_FEE*2)))/100:.2f}{txcolors.DEFAULT}")
-
+        if (SELL_AFTER_DURATION > 0 and HoldTime >= SELL_AFTER_DURATION) or LastPrice < SL or LastPrice > TP and not USE_TRAILING_STOP_LOSS:
+            if LastPrice < SL:
+                print(f"{txcolors.SELL_PROFIT if PriceChange >= 0. else txcolors.SELL_LOSS}SL reached, selling {coins_bought[coin]['volume']} {coin} - {BuyPrice} - {LastPrice} : {PriceChange-(TRADING_FEE*2):.2f}% Est:${(QUANTITY*(PriceChange-(TRADING_FEE*2)))/100:.2f} Held for {int(HoldTime)} minutes {txcolors.DEFAULT}")
+            if LastPrice > TP:
```
Comment (getsec) [inline Binance Detect Moonings.py:386]: Why would you want to sell during take profit. Also, this functionality will ONLY execute if TSL is off. Almost everyone uses TSL...

File: config.yml [hunk 2]
```diff
@@ -51,6 +51,9 @@ trading_options:
   # define in % when to take profit on a profitable coin.
   TAKE_PROFIT: .8
 
+  # Sell coin after a fixed duration in minutes. Value 0 means this fuctinality is disables.
+  SELL_AFTER_DURATION: 0
```
Comment (getsec) [inline config.yml:55]: Inherit 0s should not reflect disabled. Use something like
```yml
SELL_AFTER_TIME: True,False
SELL_AFTER_DURATION: 10
```

IMHO, this shouldnt be in this config or even in the binance script. It should be its own module which users can enable if they want...

**Review by getsec — CHANGES_REQUESTED** [review]
Honestly, while i think this is `neat` functionality, I dont think most of our user base would use this due to the fact it requires TRAILING_STOP_LOSS to be disabled. I'd request you make this a module that can be loaded in if the user wishes to use this... However, that will require an entirely different PR.

**Comment by yaka996** [comment]
Thanks for having reviewed this.
I'll try to improve this based on your comments and resubmit when I have something ready.
Have a nice day!

**Pull request closed without merging**